src/date-input/date-input.tsx

```diff
@@ -161,6 +161,10 @@ export const DateInput = ({
         if (onChange) {
             onChange(changeValue);
         }
+        // clear hover value for mobile when onMouseLeave={handleMouseLeaveCell} is not triggered due to touch input
+        if (!withButton && hoveredDate) {
+            setHoveredDate(undefined);
+        }
```

Comment:
any specific reason for putting this in `performOnChangeHandler` vs `handleSelect`?

Reply:
no specific reason, i can move it to `handleSelect` too

Reply:
ok, would suggest to move this logic out to keep `performOnChangeHandler` single-purpose

Reply:
done